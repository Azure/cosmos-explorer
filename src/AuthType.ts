export enum AuthType {
  AAD = "aad",
  EncryptedToken = "encryptedtoken",
  MasterKey = "masterkey",
  ResourceToken = "resourcetoken",
  ConnectionString = "connectionstring"
}
