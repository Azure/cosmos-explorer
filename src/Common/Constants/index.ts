import * as AfecFeatures from "./AfecFeatures";
import * as AnalyticalStorageTtl from "./AnalyticalStorageTtl";
import * as ApiEndpoints from "./ApiEndpoints";
import * as ApiType from "./ApiType";
import * as Areas from "./Areas";
import * as ArmApiVersions from "./ArmApiVersions";
import * as ArmResourceTypes from "./ArmResourceTypes";
import * as BackendDefaults from "./BackendDefaults";
import * as CapabilityNames from "./CapabilityNames";
import * as CassandraBackend from "./CassandraBackend";
import * as ClientDefaults from "./ClientDefaults";
import * as CodeOfConductEndpoints from "./CodeOfConductEndpoints";
import * as ConfigurationOverridesValues from "./ConfigurationOverridesValues";
import * as CorrelationBackend from "./CorrelationBackend";
import * as DocumentsGridMetrics from "./DocumentsGridMetrics";
import * as EndpointsRegex from "./EndpointsRegex";
import * as Flights from "./Flights";
import * as HashRoutePrefixes from "./HashRoutePrefixes";
import * as HttpHeaders from "./HttpHeaders";
import * as HttpStatusCodes from "./HttpStatusCodes";
import * as KeyCodes from "./KeyCodes";
import * as MongoDBAccounts from "./MongoDBAccounts";
import * as NormalizedEventKey from "./NormalizedEventKey";
import * as Notebook from "./Notebook";
import * as OfferVersions from "./OfferVersions";
import * as Queries from "./Queries";
import * as SavedQueries from "./SavedQueries";
import * as ServerIds from "./ServerIds";
import * as SparkLibrary from "./SparkLibrary";
import * as TagNames from "./TagNames";
import * as TerminalQueryParams from "./TerminalQueryParams";
import * as TryCosmosExperience from "./TryCosmosExperience";
import * as Urls from "./Urls";

const StyleConstants = require("less-vars-loader!../../less/Common/Constants.less");

export {
  StyleConstants,
  SparkLibrary,
  ConfigurationOverridesValues,
  OfferVersions,
  AnalyticalStorageTtl,
  Notebook,
  TryCosmosExperience,
  NormalizedEventKey,
  KeyCodes,
  HashRoutePrefixes,
  Urls,
  HttpStatusCodes,
  ApiType,
  HttpHeaders,
  Areas,
  DocumentsGridMetrics,
  SavedQueries,
  Queries,
  CassandraBackend,
  MongoDBAccounts,
  TagNames,
  AfecFeatures,
  Flights,
  CorrelationBackend,
  CapabilityNames,
  ClientDefaults,
  BackendDefaults,
  ArmResourceTypes,
  ArmApiVersions,
  TerminalQueryParams,
  CodeOfConductEndpoints,
  ApiEndpoints,
  EndpointsRegex,
  ServerIds,
};

export enum ConflictOperationType {
  Replace = "replace",
  Create = "create",
  Delete = "delete",
}

export const EmulatorMasterKey =
  //[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="Well known public masterKey for emulator")]
  "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";

// A variable @MyVariable defined in Constants.less is accessible as StyleConstants.MyVariable

export enum AccountKind {
  DocumentDB = "DocumentDB",
  MongoDB = "MongoDB",
  Parse = "Parse",
  GlobalDocumentDB = "GlobalDocumentDB",
  Default = "DocumentDB",
}

export enum MongoBackendEndpointType {
  local,
  remote,
}
