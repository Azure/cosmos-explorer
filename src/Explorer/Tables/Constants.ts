export var TableType = {
  String: "String",
  Boolean: "Boolean",
  Binary: "Binary",
  DateTime: "DateTime",
  Double: "Double",
  Guid: "Guid",
  Int32: "Int32",
  Int64: "Int64",
};

export var CassandraType = {
  Ascii: "Ascii",
  Bigint: "Bigint",
  Blob: "Blob",
  Boolean: "Boolean",
  Decimal: "Decimal",
  Double: "Double",
  Float: "Float",
  Int: "Int",
  Text: "Text",
  Uuid: "Uuid",
  Varchar: "Varchar",
  Varint: "Varint",
  Inet: "Inet",
  Smallint: "Smallint",
  Tinyint: "Tinyint",
};

export var ClauseRule = {
  And: "And",
  Or: "Or",
};

export var Operator = {
  EqualTo: "==",
  GreaterThan: ">",
  GreaterThanOrEqualTo: ">=",
  LessThan: "<",
  LessThanOrEqualTo: "<=",
  NotEqualTo: "<>",
  Equal: "=",
};

export var ODataOperator = {
  EqualTo: "eq",
  GreaterThan: "gt",
  GreaterThanOrEqualTo: "ge",
  LessThan: "lt",
  LessThanOrEqualTo: "le",
  NotEqualTo: "ne",
};

export var timeOptions = {
  lastHour: "Last hour",
  last24Hours: "Last 24 hours",
  last7Days: "Last 7 days",
  last31Days: "Last 31 days",
  last365Days: "Last 365 days",
  currentMonth: "Current month",
  currentYear: "Current year",
  custom: "Custom...",
};

export var htmlSelectors = {
  dataTableSelector: "#storageTable",
  dataTableAllRowsSelector: "#storageTable tbody tr",
  dataTableHeadRowSelector: ".dataTable thead tr",
  dataTableBodyRowSelector: ".dataTable tbody tr",
  dataTableScrollBodySelector: ".dataTables_scrollBody",
  dataTableScrollContainerSelector: ".dataTables_scroll",
  dataTableHeaderTypeSelector: "table thead th",
  dataTablePaginationButtonSelector: ".paginate_button",
  dataTableHeaderTableSelector: "#storageTable_wrapper .dataTables_scrollHeadInner table",
  dataTableBodyTableSelector: "#storageTable_wrapper .dataTables_scrollBody table",
  searchInputField: ".search-input",
  uploadDropdownSelector: "#upload-dropdown",
  navigationDropdownSelector: "#navigation-dropdown",
  addressBarInputSelector: "#address-bar",
  breadCrumbsSelector: "#breadcrumb-list",
  breadCrumbItemsSelector: ".breadcrumb li a",
  paginateSelector: "#storageTable_paginate",
  dataTablesInfoSelector: "#storageTable_info",
  selectAllDropdownSelector: "#select-all-dropdown",
};

export var defaultHeader = " ";

export var EntityKeyNames = {
  PartitionKey: "PartitionKey",
  RowKey: "RowKey",
  Timestamp: "Timestamp",
  Metadata: ".metadata",
  Etag: "etag",
};

export var htmlAttributeNames = {
  dataTableNameAttr: "name_attr",
  dataTableContentTypeAttr: "contentType_attr",
  dataTableSnapshotAttr: "snapshot_attr",
  dataTableRowKeyAttr: "rowKey_attr",
  dataTableMessageIdAttr: "messageId_attr",
  dataTableHeaderIndex: "data-column-index",
};

export var cssColors = {
  commonControlsButtonActive: "#B4C7DC" /* A darker shade of [{common-controls-button-hover-background}] */,
};

export var clauseGroupColors = ["#ffe1ff", "#fffacd", "#f0ffff", "#ffefd5", "#f0fff0"];
export var transparentColor = "transparent";

export var keyCodes = {
  RightClick: 3,
  Enter: 13,
  Esc: 27,
  Tab: 9,
  LeftArrow: 37,
  UpArrow: 38,
  RightArrow: 39,
  DownArrow: 40,
  Delete: 46,
  A: 65,
  B: 66,
  C: 67,
  D: 68,
  E: 69,
  F: 70,
  G: 71,
  H: 72,
  I: 73,
  J: 74,
  K: 75,
  L: 76,
  M: 77,
  N: 78,
  O: 79,
  P: 80,
  Q: 81,
  R: 82,
  S: 83,
  T: 84,
  U: 85,
  V: 86,
  W: 87,
  X: 88,
  Y: 89,
  Z: 90,
  Period: 190,
  DecimalPoint: 110,
  F1: 112,
  F2: 113,
  F3: 114,
  F4: 115,
  F5: 116,
  F6: 117,
  F7: 118,
  F8: 119,
  F9: 120,
  F10: 121,
  F11: 122,
  F12: 123,
  Dash: 189,
};

export var InputType = {
  Text: "text",
  // Chrome doesn't support datetime, instead, datetime-local is supported.
  DateTime: "datetime-local",
  Number: "number",
};
