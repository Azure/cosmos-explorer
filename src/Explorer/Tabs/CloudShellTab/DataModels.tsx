/**
 * Copyright (c) Microsoft Corporation.  All rights reserved.
 */

export const enum OsType {
    Linux = "linux",
    Windows = "windows"
}

export const enum ShellType {
    Bash = "bash",
    PowerShellCore = "pwsh"
}
  
export const enum NetworkType {
    Default = "Default",
    Isolated = "Isolated"
}

export const enum SessionType {
    Mounted = "Mounted",
    Ephemeral = "Ephemeral"
}

export type Settings = {
    location: string;
    sessionType: SessionType;
    osType: OsType;
};

export type ProvisionConsoleResponse = {
    properties: {
        osType: OsType;
        provisioningState: string;
        uri: string;
    };
};

export type Authorization = {
    token: string;
};

export type ConnectTerminalResponse = {
    id: string;
    idleTimeout: string;
    rootDirectory: string;
    socketUri: string;
    tokenUpdated: boolean;
};


