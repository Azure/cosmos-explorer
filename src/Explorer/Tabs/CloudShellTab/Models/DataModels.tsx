export const enum OsType {
  Linux = "linux",
  Windows = "windows",
}

export const enum ShellType {
  Bash = "bash",
  PowerShellCore = "pwsh",
}

export const enum NetworkType {
  Default = "Default",
  Isolated = "Isolated",
}

/**
 * Azure CloudShell session types:
 * - Mounted: Sessions with persistent storage via an Azure File Share mount.
 *   Files and configurations are preserved between sessions, allowing for
 *   continuity of work across multiple CloudShell sessions.
 *
 * - Ephemeral: Temporary sessions without persistent storage.
 *   All files and changes are discarded when the session ends.
 *   These sessions start faster but don't retain user data.
 *
 * The session type affects resource allocation, startup time,
 * and whether user files/configurations persist between sessions.
 */
export const enum SessionType {
  Mounted = "Mounted",
  Ephemeral = "Ephemeral",
}

export type CloudShellSettings = {
  properties: UserSettingProperties;
};

export type UserSettingProperties = {
  networkType: string;
  preferredLocation: string;
  preferredOsType: OsType;
  preferredShellType: ShellType;
  userSubscription: string;
  sessionType: SessionType;
  vnetSettings: object;
};

export type ProvisionConsoleResponse = {
  properties: {
    osType: OsType;
    provisioningState: string;
    uri: string;
  };
};

export type Authorization = {
  token: string;
};

export type ConnectTerminalResponse = {
  id: string;
  idleTimeout: string;
  rootDirectory: string;
  socketUri: string;
  tokenUpdated: boolean;
};

export type ProviderAuthorization = {
  applicationId: string;
  roleDefinitionId: string;
};

export type ProviderResourceType = {
  resourceType: string;
  locations: string[];
  apiVersions: string[];
  defaultApiVersion?: string;
  capabilities?: string;
};

export type RegistrationState = "Registered" | "NotRegistered" | "Registering" | "Unregistering";
export type RegistrationPolicy = "RegistrationRequired" | "RegistrationOptional";

export type CloudShellProviderInfo = {
  id: string;
  namespace: string;
  authorizations?: ProviderAuthorization[];
  resourceTypes: ProviderResourceType[];
  registrationState: RegistrationState;
  registrationPolicy: RegistrationPolicy;
};
