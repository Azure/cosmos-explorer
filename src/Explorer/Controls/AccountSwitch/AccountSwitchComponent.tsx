import { AuthType } from "../../../AuthType";
import { StyleConstants } from "../../../Common/Constants";
import { DatabaseAccount, Subscription } from "../../../Contracts/DataModels";

import * as React from "react";
import { DefaultButton, IButtonStyles, IButtonProps } from "office-ui-fabric-react/lib/Button";
import { IContextualMenuProps } from "office-ui-fabric-react/lib/ContextualMenu";
import { Dropdown, IDropdownOption, IDropdownProps } from "office-ui-fabric-react/lib/Dropdown";

export interface AccountSwitchComponentProps {
  authType: AuthType;
  selectedAccountName: string;
  accounts: DatabaseAccount[];
  isLoadingAccounts: boolean;
  onAccountChange: (newAccount: DatabaseAccount) => void;
  selectedSubscriptionId: string;
  subscriptions: Subscription[];
  isLoadingSubscriptions: boolean;
  onSubscriptionChange: (newSubscription: Subscription) => void;
  displayText?: string;
}

export class AccountSwitchComponent extends React.Component<AccountSwitchComponentProps> {
  public render(): JSX.Element {
    return this.props.authType === AuthType.AAD ? this._renderSwitchDropDown() : this._renderAccountName();
  }

  private _renderSwitchDropDown(): JSX.Element {
    const { displayText, selectedAccountName } = this.props;

    const menuProps: IContextualMenuProps = {
      directionalHintFixed: true,
      className: "accountSwitchContextualMenu",
      items: [
        {
          key: "switchSubscription",
          onRender: this._renderSubscriptionDropdown.bind(this)
        },
        {
          key: "switchAccount",
          onRender: this._renderAccountDropDown.bind(this)
        }
      ]
    };

    const buttonStyles: IButtonStyles = {
      root: {
        fontSize: StyleConstants.DefaultFontSize,
        height: 40,
        padding: 0,
        paddingLeft: 10,
        marginRight: 5,
        backgroundColor: StyleConstants.BaseDark,
        color: StyleConstants.BaseLight
      },
      rootHovered: {
        backgroundColor: StyleConstants.BaseHigh,
        color: StyleConstants.BaseLight
      },
      rootFocused: {
        backgroundColor: StyleConstants.BaseHigh,
        color: StyleConstants.BaseLight
      },
      rootPressed: {
        backgroundColor: StyleConstants.BaseHigh,
        color: StyleConstants.BaseLight
      },
      rootExpanded: {
        backgroundColor: StyleConstants.BaseHigh,
        color: StyleConstants.BaseLight
      },
      textContainer: {
        flexGrow: "initial"
      }
    };

    const buttonProps: IButtonProps = {
      text: displayText || selectedAccountName,
      menuProps: menuProps,
      styles: buttonStyles,
      className: "accountSwitchButton",
      id: "accountSwitchButton"
    };

    return <DefaultButton {...buttonProps} />;
  }

  private _renderSubscriptionDropdown(): JSX.Element {
    const { subscriptions, selectedSubscriptionId, isLoadingSubscriptions } = this.props;
    const options: IDropdownOption[] = subscriptions.map(sub => {
      return {
        key: sub.subscriptionId,
        text: sub.displayName,
        data: sub
      };
    });

    const placeHolderText = isLoadingSubscriptions
      ? "Loading subscriptions"
      : !options || !options.length
      ? "No subscriptions found in current directory"
      : "Select subscription from list";

    const dropdownProps: IDropdownProps = {
      label: "Subscription",
      className: "accountSwitchSubscriptionDropdown",
      options: options,
      onChange: this._onSubscriptionDropdownChange,
      defaultSelectedKey: selectedSubscriptionId,
      placeholder: placeHolderText,
      styles: {
        callout: "accountSwitchSubscriptionDropdownMenu"
      }
    };

    return <Dropdown {...dropdownProps} />;
  }

  private _onSubscriptionDropdownChange = (e: React.FormEvent<HTMLDivElement>, option: IDropdownOption): void => {
    if (!option) {
      return;
    }

    this.props.onSubscriptionChange(option.data);
  };

  private _renderAccountDropDown(): JSX.Element {
    const { accounts, selectedAccountName, isLoadingAccounts } = this.props;
    const options: IDropdownOption[] = accounts.map(account => {
      return {
        key: account.name,
        text: account.name,
        data: account
      };
    });
    // Fabric UI will also try to select the first non-disabled option from dropdown.
    // Add a option to prevent pop the message when user click on dropdown on first time.
    options.unshift({
      key: "select from list",
      text: "Select Cosmos DB account from list",
      data: undefined
    });

    const placeHolderText = isLoadingAccounts
      ? "Loading Cosmos DB accounts"
      : !options || !options.length
      ? "No Cosmos DB accounts found"
      : "Select Cosmos DB account from list";

    const dropdownProps: IDropdownProps = {
      label: "Cosmos DB Account Name",
      className: "accountSwitchAccountDropdown",
      options: options,
      onChange: this._onAccountDropdownChange,
      defaultSelectedKey: selectedAccountName,
      placeholder: placeHolderText,
      styles: {
        callout: "accountSwitchAccountDropdownMenu"
      }
    };

    return <Dropdown {...dropdownProps} />;
  }

  private _onAccountDropdownChange = (e: React.FormEvent<HTMLDivElement>, option: IDropdownOption): void => {
    if (!option) {
      return;
    }

    this.props.onAccountChange(option.data);
  };

  private _renderAccountName(): JSX.Element {
    const { displayText, selectedAccountName } = this.props;
    return <span className="accountNameHeader">{displayText || selectedAccountName}</span>;
  }
}
