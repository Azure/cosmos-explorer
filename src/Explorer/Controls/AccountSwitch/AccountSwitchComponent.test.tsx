import React from "react";
import { shallow, mount } from "enzyme";
import { AccountSwitchComponent, AccountSwitchComponentProps } from "./AccountSwitchComponent";
import { AuthType } from "../../../AuthType";
import { DatabaseAccount, Subscription } from "../../../Contracts/DataModels";
import { AccountKind } from "../../../Common/Constants";

const createBlankProps = (): AccountSwitchComponentProps => {
  return {
    authType: null,
    displayText: "",
    accounts: [],
    selectedAccountName: null,
    isLoadingAccounts: false,
    onAccountChange: jest.fn(),
    subscriptions: [],
    selectedSubscriptionId: null,
    isLoadingSubscriptions: false,
    onSubscriptionChange: jest.fn()
  };
};

const createBlankAccount = (): DatabaseAccount => {
  return {
    id: "",
    kind: AccountKind.Default,
    name: "",
    properties: null,
    location: "",
    tags: null,
    type: ""
  };
};

const createBlankSubscription = (): Subscription => {
  return {
    subscriptionId: "",
    displayName: "",
    authorizationSource: "",
    state: "",
    subscriptionPolicies: null,
    tenantId: "",
    uniqueDisplayName: ""
  };
};

const createFullProps = (): AccountSwitchComponentProps => {
  const props = createBlankProps();
  props.authType = AuthType.AAD;
  const account1 = createBlankAccount();
  account1.name = "account1";
  const account2 = createBlankAccount();
  account2.name = "account2";
  const account3 = createBlankAccount();
  account3.name = "superlongaccountnamestringtest";
  props.accounts = [account1, account2, account3];
  props.selectedAccountName = "account2";

  const sub1 = createBlankSubscription();
  sub1.displayName = "sub1";
  sub1.subscriptionId = "a6062a74-5d53-4b20-9545-000b95f22297";
  const sub2 = createBlankSubscription();
  sub2.displayName = "subsubsubsubsubsubsub2";
  sub2.subscriptionId = "b20b3e93-0185-4326-8a9c-d44bac276b6b";
  props.subscriptions = [sub1, sub2];
  props.selectedSubscriptionId = "a6062a74-5d53-4b20-9545-000b95f22297";

  return props;
};

describe("test render", () => {
  it("renders no auth type -> handle error in code", () => {
    const props = createBlankProps();

    const wrapper = shallow(<AccountSwitchComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  // Encrypted Token
  it("renders auth security token, with selected account name", () => {
    const props = createBlankProps();
    props.authType = AuthType.EncryptedToken;
    props.selectedAccountName = "testaccount";

    const wrapper = shallow(<AccountSwitchComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  // AAD
  it("renders auth aad, with all information", () => {
    const props = createFullProps();
    const wrapper = shallow(<AccountSwitchComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders auth aad all dropdown menus", () => {
    const props = createFullProps();
    const wrapper = mount(<AccountSwitchComponent {...props} />);

    expect(wrapper.exists("div.accountSwitchContextualMenu")).toBe(false);
    wrapper.find("button.accountSwitchButton").simulate("click");
    expect(wrapper.exists("div.accountSwitchContextualMenu")).toBe(true);

    expect(wrapper.exists("div.accountSwitchSubscriptionDropdown")).toBe(true);
    wrapper.find("DropdownBase.accountSwitchSubscriptionDropdown").simulate("click");
    // Click will dismiss the first contextual menu in enzyme. Need to dig deeper to achieve below test

    // expect(wrapper.exists("div.accountSwitchSubscriptionDropdownMenu")).toBe(true);
    // expect(wrapper.find("button.ms-Dropdown-item").length).toBe(2);
    // wrapper.find("div.accountSwitchSubscriptionDropdown").simulate("click");
    // expect(wrapper.exists("div.accountSwitchSubscriptionDropdownMenu")).toBe(false);

    // expect(wrapper.exists("div.accountSwitchAccountDropdown")).toBe(true);
    // wrapper.find("div.accountSwitchAccountDropdown").simulate("click");
    // expect(wrapper.exists("div.accountSwitchAccountDropdownMenu")).toBe(true);
    // expect(wrapper.find("button.ms-Dropdown-item").length).toBe(3);
    // wrapper.find("div.accountSwitchAccountDropdown").simulate("click");
    // expect(wrapper.exists("div.accountSwitchAccountDropdownMenu")).toBe(false);

    // wrapper.find("button.accountSwitchButton").simulate("click");
    // expect(wrapper.exists("div.accountSwitchContextualMenu")).toBe(false);

    wrapper.unmount();
  });
});

// describe("test function", () => {
//   it("switch subscription function", () => {
//     const props = createFullProps();
//     const wrapper = mount(<AccountSwitchComponent {...props} />);

//     wrapper.find("button.accountSwitchButton").simulate("click");
//     wrapper.find("div.accountSwitchSubscriptionDropdown").simulate("click");
//     wrapper
//       .find("button.ms-Dropdown-item")
//       .at(1)
//       .simulate("click");
//     expect(props.onSubscriptionChange).toBeCalled();
//     expect(props.onSubscriptionChange).toHaveBeenCalled();

//     wrapper.unmount();
//   });

//   it("switch account", () => {
//     const props = createFullProps();
//     const wrapper = mount(<AccountSwitchComponent {...props} />);

//     wrapper.find("button.accountSwitchButton").simulate("click");
//     wrapper.find("div.accountSwitchAccountDropdown").simulate("click");
//     wrapper
//       .find("button.ms-Dropdown-item")
//       .at(0)
//       .simulate("click");
//     expect(props.onAccountChange).toBeCalled();
//     expect(props.onAccountChange).toHaveBeenCalled();

//     wrapper.unmount();
//   });
// });
