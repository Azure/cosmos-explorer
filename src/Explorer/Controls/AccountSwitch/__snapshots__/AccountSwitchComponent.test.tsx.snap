// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render renders auth aad, with all information 1`] = `
<CustomizedDefaultButton
  className="accountSwitchButton"
  id="accountSwitchButton"
  menuProps={
    Object {
      "className": "accountSwitchContextualMenu",
      "directionalHintFixed": true,
      "items": Array [
        Object {
          "key": "switchSubscription",
          "onRender": [Function],
        },
        Object {
          "key": "switchAccount",
          "onRender": [Function],
        },
      ],
    }
  }
  styles={
    Object {
      "root": Object {
        "backgroundColor": undefined,
        "color": undefined,
        "fontSize": undefined,
        "height": 40,
        "marginRight": 5,
        "padding": 0,
        "paddingLeft": 10,
      },
      "rootExpanded": Object {
        "backgroundColor": undefined,
        "color": undefined,
      },
      "rootFocused": Object {
        "backgroundColor": undefined,
        "color": undefined,
      },
      "rootHovered": Object {
        "backgroundColor": undefined,
        "color": undefined,
      },
      "rootPressed": Object {
        "backgroundColor": undefined,
        "color": undefined,
      },
      "textContainer": Object {
        "flexGrow": "initial",
      },
    }
  }
  text="account2"
/>
`;

exports[`test render renders auth security token, with selected account name 1`] = `
<span
  className="accountNameHeader"
>
  testaccount
</span>
`;

exports[`test render renders no auth type -> handle error in code 1`] = `
<span
  className="accountNameHeader"
/>
`;
